### PR TITLE
device: implement std::error::Error for VfioError

### PR DESCRIPTION
**Pull request by jiangliu**
Signed-off-by: Liu Jiang <gerry@linux.alibaba.com>

### CODE REVIEW

File: src/vfio_device.rs [hunk 1]
```diff
@@ -58,12 +58,8 @@ pub type Result<T> = std::result::Result<T, VfioError>;
 impl fmt::Display for VfioError {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         match self {
-            VfioError::OpenContainer(e) => {
-                write!(f, "failed to open /dev/vfio/vfio container: {}", e)
-            }
-            VfioError::OpenGroup(e, ref p) => {
-                write!(f, "failed to open /dev/vfio/{} group: {}", p, e)
-            }
+            VfioError::OpenContainer(_e) => write!(f, "failed to open /dev/vfio/vfio container"),
```
Comment (sameo) [inline src/vfio_device.rs:61]: Why not:
```Rust
VfioError::OpenContainer(e) => write!(f, "failed to open /dev/vfio/vfio container: {}", e.source().unwrap()),
```

Comment (jiangliu) [reply src/vfio_device.rs:61]: There's discussion about this topic here: https://github.com/rust-vmm/event-manager/pull/28#pullrequestreview-448709021

Basically we hope to follow the standard std::error::Error design pattern.

Comment (sameo) [reply src/vfio_device.rs:61]: So from the VMM code, if I get a VFIO error, I would have to implement the following pattern to get the whole chain:

```Rust
error!("VFIO error {:?}", e.source().unwrap())
```

Or use the `anyhow` crate to handle it for me?

Comment (jiangliu) [reply src/vfio_device.rs:61]: anyhow is recommended, and cloud hypervisor already uses anyhow anyway.
It's really hard to compose the error message when the error has a deep chain.

**Review by sameo — APPROVED** [review]

**Pull request merged**